src/TraceEvent/TraceEvent.cs

```diff
@@ -397,6 +398,8 @@ string ITraceParserServices.ProviderNameForGuid(Guid taskOrProviderGuid)
         internal /*protected*/ DynamicTraceEventParser _Dynamic;
         internal /*protected*/ RegisteredTraceEventParser _Registered;
 #endif //  !NOT_WINDOWS
+
+        internal /*protected*/ string origin;
```

Comment:
I don't want this string because it is relative inefficient to test for it (it is a string comparison).   I would prefer that you simply test with a type test (that is 'source is EventPIpeSource'), for ugly ad hoc things or put in a real virtual method if we think we new abstractions (see comments at use points).